xbmc/dialogs/GUIDialogPlayEject.cpp

```diff
@@ -89,10 +106,33 @@ void CGUIDialogPlayEject::OnInitWindow()
 bool CGUIDialogPlayEject::ShowAndGetInput(const CFileItem & item,
   unsigned int uiAutoCloseTime /* = 0 */)
 {
-  // Make sure we're actually dealing with a Disc Stub
-  if (!item.IsDiscStub())
+  // Make sure we're actually dealing with a Stub
+  if (!item.IsEFileStub() && !item.IsDiscStub())
     return false;
 
+  CURL stubUrl = item.GetURL();
+  isDisc = item.IsDiscStub();
```

Comment:
I did this to avoid several checks in the dialog. The alternative is check everytime with the class or to create a separate dialog for efile and disc stubs. Probably the class needs a rename since it's there only for disc stubs and now is needed for both